doc: update test concurrency description / default value (#46457)
nodejs/node: doc/api/test.md

  properties are supported:
  * `concurrency` {number|boolean} If a number is provided,
    then that many files would run in parallel.
    If `true`, it would run `os.availableParallelism() - 1` files in parallel.
    If `false`, it would only run one test at a time.

>     If unspecified, subtests inherit this value from their parent.

Reviewer: AFAIK it can never be a subtest.
```suggestion
```

Author: The way that I read this is: when the option isn't passed to `test`, any `t.test` within `test` will inherit that value. 

The same text is also used in the `options.timeout` description and a few other places in this doc.

  properties are supported:
  * `concurrency` {number|boolean} If a number is provided,
    then that many files would run in parallel.
    If truthy, it would run (number of cpu cores - 1)
    files in parallel.
    If falsy, it would only run one file at a time.
    If unspecified, subtests inherit this value from their parent.

>     **Default:** `true`.

Author: Ah, yes - correct. This one was actually wrong. The default is `false/1`, not `true`.

>     If `true`, it would run `os.availableParallelism() - 1` tests in parallel.
>     If `false`, it would only run one test at a time.

Reviewer 2: ```suggestion
    If `true`, it would run `os.availableParallelism() - 1` test files in parallel.
    If `false`, it would only run one test file at a time.
```

  properties are supported:
  * `concurrency` {number|boolean} If a number is provided,
    then that many tests would run in parallel.

>     If `true`, it would run `os.availableParallelism() - 1` files in parallel.

Reviewer: ```suggestion
    If `true`, it would run `os.availableParallelism() - 1` tests in parallel.
```

  properties are supported:
  * `concurrency` {number|boolean} If a number is provided,
    then that many tests would run in parallel.

>     If `true`, it would run `os.availableParallelism() - 1` files in parallel.
>     tests in parallel.

Reviewer: ```suggestion
    If `true`, it would run `os.availableParallelism() - 1` tests in parallel.
```

    For subtests, it will be `Infinity` tests in parallel.

>     If `false`, it would only run one file at a time.

Reviewer: ```suggestion
    If `false`, it would only run one test at a time.
```

    For subtests, it will be `Infinity` tests in parallel.
    If falsy, it would only run one test at a time.
    If unspecified, subtests inherit this value from their parent.

>     **Default:** `1`.

Reviewer: `false` and `1` are strictly equivalent, I think it's better to document `false` as I expect most users to only use a boolean value for this field. No strong feeling though.

Author: I can revert this although `test` and `context.text` will have different default values, unless that's not an issue?

Reviewer: Consistency would be better, so I'd say we should put `false` for both, but also I'm not feeling strongly either way.

Reviewer 2: +1 for false

Author: Sure, I've pushed an update that reverts this change but updates the `test` concurrency option so that the two are consistent.

  `fn` does not have a name.
* `options` {Object} Configuration options for the subtest. The following
  properties are supported:

>   * `concurrency` {number|boolean} If a number is provided,

Reviewer: ```suggestion
  * `concurrency` {number|boolean|null} If a number is provided,
```

>     then that many files would run in parallel.

Reviewer: ```suggestion
    then that many tests would run in parallel.
```

>     If `true`, it would run `os.availableParallelism() - 1` files in parallel.

Reviewer: For subtests, we use `infinity`
```suggestion
    If `true`, it would run `Infinity` tests in parallel.
```

    If `false`, it would only run one test at a time.
    If unspecified, subtests inherit this value from their parent.

>     **Default:** `false`.

Reviewer: ```suggestion
    **Default:** `null`.
```

Reviewer: Isn’t it because the parent has a concurrency of `1`/`false`?

Author: If the parent has concurrency turned off by default via `1`/`false` then the subtest should also have that value, right?

Reviewer: The fact that the default value for the parent is `false` is irrelevant imo, we are not documenting what the actual value will be if only defaults are being used, we are describing the API behavior. Wouldn't you agree that it would be confusing for someone that's using something that's not `1` in the parent test? If we say in the docs that `null` means "inherit the value from the parent", then the default value is `null`, not `false` nor `1`. Does that make sense?

Author: Sure, would this logic also apply to the other places where values are inherited from the parent? like `options.timeout` in the various places?

Reviewer: Yes I think that would apply the same. I suspect the current text is a result of a copy pasting and the mistake slip through. We can either keep this PR focused on `concurrency` and open a PR in parallel to fix the other options, or do that in this PR and change the commit message.

Author: Yes, I'll make the changes you've suggested here and keep this PR focused on `concurrency`. I'll create a follow up PR to fix the other options.

  `fn` does not have a name.
* `options` {Object} Configuration options for the subtest. The following
  properties are supported:
  * `concurrency` {number|boolean} If a number is provided,
    then that many files would run in parallel.
    If `true`, it would run `os.availableParallelism() - 1` files in parallel.

>     files in parallel.

Reviewer: ```suggestion
```

>     If truthy, it would run (number of cpu cores - 1)
>     files in parallel.
>     If falsy, it would only run one file at a time.

Reviewer: We should use this PR to update this: `1` is truthy, yet it doesn't mean that we would run more than one test at a time. Also, we are no longer using the number of CPU cores.
```suggestion
    If `true`, it would run `os.availableParallelism()  - 1` files in parallel.
    If `false`, it would only run one file at a time.
```

Author: Sure, I've updated both `test` and `context.test` with your suggestion as they were both using the same description.
